src/providers/node.rs

```diff
@@ -83,9 +120,19 @@ impl Provider for NodeProvider {
 
     fn build(&self, app: &App, _env: &Environment) -> Result<Option<BuildPhase>> {
         let mut build_phase = BuildPhase::default();
-
-        if NodeProvider::has_script(app, "build")? {
-            let pkg_manager = NodeProvider::get_package_manager(app);
+        let pkg_manager = NodeProvider::get_package_manager(app);
+
+        if NodeProvider::is_nx_monorepo(app) {
+            let app_name = NodeProvider::get_nx_app_name(app, _env)?.unwrap();
+            build_phase.add_cmd(format!(
+                "{} run build {} --configuration=production",
+                pkg_manager, app_name
+            ));
+            // Remove the package.json so when we run node /dist/**/file.js node does
+            // not look at the package.json and infer things it should not, Such as wether its a es module or not.
+            build_phase.add_cmd("rm /app/package.json".to_owned());
+            build_phase.add_cache_directory(NodeProvider::get_nx_output_path(app, _env).unwrap());
```

Comment:
Should I be caching the output folder of the build?

Reply:
The cached directories do not appear in the final image so it wouldn't work to cache the build output directory.

Reply:
Makes sense, deleted this